packages/driver-adapter-utils/src/binder.ts

```diff
@@ -16,11 +16,34 @@ import type {
 
 const debug = Debug('driver-adapter-utils')
 
+const DRIVER_ADAPTER_EXTERNAL_ERROR = 'P2036'
+
 class ErrorRegistryInternal implements ErrorRegistry {
   private registeredErrors: ErrorRecord[] = []
 
-  consumeError(id: number): ErrorRecord | undefined {
-    return this.registeredErrors[id]
+  lookupError(
+    error: number | { error_code: string; meta?: { id: number } } | { message: string },
+  ): ErrorRecord | undefined {
+    if (typeof error === 'number') {
+      const errorRecord = this.registeredErrors[error]
+      if (!errorRecord) throw `External error with reported id was not registered`
```

Comment:
I feel like the logic is backwards compared to the function signature. We throw if the specified error is not found (although the signature suggests `undefined` would be returned in this case) and only return `undefined` in the final `else` branch if the shape of the input was completely wrong and we couldn't interpret it (which it would be more appropriate to throw an error about).